### PR TITLE
Allow clang-format version to be set

### PR DESCRIPTION
**Pull request by wxmerkt**
I've tested this on our build server and it works (if the set version exists and can be apt-get installed, of course).

- Relates to #339 

cc: @ipa-mdl

### CODE REVIEW

File: doc/index.rst [hunk 1]
```diff
@@ -177,7 +177,7 @@ Note that some of these currently tied only to a single option, but we still lea
...
 * **CCACHE_DIR** (default: not set): If set, `ccache <https://en.wikipedia.org/wiki/Ccache>`_ gets enabled for your build to speed up the subsequent builds in the same job if anything. See `detail. <https://github.com/ros-industrial/industrial_ci/blob/master/doc/index.rst#cache-build-artifacts-to-speed-up-the-subsequent-builds-if-any>`_
-* **CLANG_FORMAT_CHECK** (default: not set. Value range: ``[<format-style>|file]``): If set, run the `clang-format <https://clang.llvm.org/docs/ClangFormat.html>`_ check. Set the argument to ``file`` if the style configuration should be loaded from a ``.clang-format`` file, located in one of the parent directories of the source file.
+* **CLANG_FORMAT_CHECK** (default: not set. Value range: ``[<format-style>|file]``): If set, run the `clang-format <https://clang.llvm.org/docs/ClangFormat.html>`_ check. Set the argument to ``file`` if the style configuration should be loaded from a ``.clang-format`` file, located in one of the parent directories of the source file. If a particular version of clang-format is desired, please specify via **CLANG_FORMAT_EXECUTABLE** (default: clang-format).
```
Comment (ipa-mdl) [inline doc/index.rst]: Please split both and give an example for `CLANG_FORMAT_EXECUTABLE`

File: industrial_ci/src/tests/clang_format_check.sh [hunk 2]
```diff
@@ -21,9 +21,15 @@ function run_clang_format_check() {
 
   DOCKER_IMAGE="$DOCKER_BASE_IMAGE" ici_require_run_in_docker # this script must be run in docker
 
+  # Check whether a specific version of clang-format is desired
```
Comment (ipa-mdl) [inline industrial_ci/src/tests/clang_format_check.sh]: variables names should not start with underscores.
I'd prefer:
```bash
local clang_format_executable=${CLANG_FORMAT_EXECUTABLE:-clang-format}
```

**Comment by ipa-mdl** [comment]
It might make sense to migrate one (passing) test case to indigo, which require the executable to be set.

**Comment by wxmerkt** [comment]
@ipa-mdl Thanks for the review. PR updated and passing test for Indigo added.

**Comment by ipa-mdl** [comment]
Just another idea: What do you think of a `CLANG_FORMAT_VERSION` variables?
```bash
local clang_format_executable="catkin-format${CLANG_FORMAT_VERSION:+-}$CLANG_FORMAT_VERSION"
```

**Comment by wxmerkt** [comment]
I think a ``CLANG_FORMAT_VERSIION`` variable would be workable as well, however, I see other added benefits with specifying the full executable name - users can specify other command line parameters/options beyond the default ``style=file``. Should I still go ahead and amend?

**Comment by ipa-mdl** [comment]
> I see other added benefits with specifying the full executable name - users can specify other command line parameters/options beyond the default style=file. 

This would not work because of the apt install..
It be okay to an a variable for additional options.

> Should I still go ahead and amend?

Yes, if it okay for you.
Specificying the version only would be less error-prone.

**Comment by wxmerkt** [comment]
Done! :-)

File: industrial_ci/src/tests/clang_format_check.sh [hunk 3]
```diff
@@ -22,7 +22,7 @@ function run_clang_format_check() {
   DOCKER_IMAGE="$DOCKER_BASE_IMAGE" ici_require_run_in_docker # this script must be run in docker
 
   # Check whether a specific version of clang-format is desired
-  local clang_format_executable="catkin-format${CLANG_FORMAT_VERSION:+-$CLANG_FORMAT_VERSION}"
+  local clang_format_executable="clang-format${CLANG_FORMAT_VERSION:+-$CLANG_FORMAT_VERSION}"
```
Comment (ipa-mdl) [inline industrial_ci/src/tests/clang_format_check.sh]: @wxmerkt: sry, my bad.

**Pull request merged**